mlflow/cli.py

```diff
@@ -106,12 +107,11 @@ def run(uri, entry_point, version, param_list, experiment_id, mode, cluster_spec
             git_password=git_password,
             use_conda=(not no_conda),
             storage_dir=storage_dir,
-            block=True,
+            block=mode == "local" or mode is None,
             run_id=run_id,
         )
-    except projects.ExecutionException:
-        import traceback
-        traceback.print_exc(file=sys.stderr)
+    except projects.ExecutionException as e:
+        eprint(e)
```

Comment:
Will this just print the message? It seems to me like it would still print `ExecutionException('whatever')`, which is not just we want.

Reply:
In fact we should make the exception message not contain `===` and add the `===` here.

Reply:
AFAICT only the exception message is printed - added a unit test to verify this behavior.  Also good call regarding formatting the error message in the CLI, made this update as well.